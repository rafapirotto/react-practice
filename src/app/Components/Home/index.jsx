import React from 'react';
import { useSelector } from 'react-redux';

import { MovieListContainer, PreviewContainer } from '../../Containers';

import { getMovies } from '../../Containers/PopularContainer/duck/operations';
import { getMyList } from '../../Containers/MyListContainer/duck/operations';
import { getMostViewed } from '../../Containers/MostViewedContainer/duck/operations';
import { getRecommended } from '../../Containers/RecommendedMoviesContainer/duck/operations';
import { getContinueWatching } from '../../Containers/ContinueWatchingContainer/duck/operations';

const Home = () => {
  const popularMoviesSelector = useSelector((state) => state.popularMovies);
  const myListMoviesSelector = useSelector((state) => state.myListMovies);
  const continueWatchingMoviesSelector = useSelector((state) => state.continueWatchingMovies);
  const mostViewedMoviesSelector = useSelector((state) => state.mostViewedMovies);
  const recommendedMoviesSelector = useSelector((state) => state.recommendedMovies);

  return (
    <>
      <PreviewContainer />
      <div className="mt-80">
        <MovieListContainer operation={getMyList} title="My List" selector={myListMoviesSelector} />
        <MovieListContainer
          operation={getMovies}
          title="Popular on Movy"
          selector={popularMoviesSelector}
        />
        <MovieListContainer
          operation={getContinueWatching}
          title="Continue Watching for Rafa"
          selector={continueWatchingMoviesSelector}
        />

        <MovieListContainer
          operation={getMostViewed}
          title="Most Viewed"
          selector={mostViewedMoviesSelector}
        />
        <MovieListContainer
          operation={getRecommended}
          title="Recommended movies"
          selector={recommendedMoviesSelector}
        />
        <MovieListContainer
          operation={getRecommended}
          title="Recommended movies"
          selector={recommendedMoviesSelector}
        />
        <MovieListContainer
          operation={getRecommended}
          title="Recommended movies"
          selector={recommendedMoviesSelector}
        />
      </div>
    </>
  );
};

export default Home;
