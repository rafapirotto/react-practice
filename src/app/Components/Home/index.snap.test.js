import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import Home from '.';

const middlewares = [thunk];

it('matches the snapshot', () => {
  const mockStore = configureStore(middlewares);
  const initialState = {
    popularMovies: { popularMovies: [] },
    myListMovies: { myListMovies: [] },
    continueWatchingMovies: { continueWatchingMovies: [] },
    mostViewedMovies: { mostViewedMovies: [] },
    recommendedMovies: { recommendedMovies: [] },
  };
  const store = mockStore(initialState);

  const component = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
