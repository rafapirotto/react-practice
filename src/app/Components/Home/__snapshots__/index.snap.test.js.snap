// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
Array [
  <div
    className="mb-80"
  />,
  <div
    className="mt-80"
  >
    <main>
      <div>
        <h1
          className="ml-24 text-3xl font-medium"
        >
          My List
        </h1>
      </div>
    </main>
    <main>
      <div>
        <h1
          className="ml-24 text-3xl font-medium"
        >
          Popular on Movy
        </h1>
      </div>
    </main>
    <main>
      <div>
        <h1
          className="ml-24 text-3xl font-medium"
        >
          Continue Watching for Rafa
        </h1>
      </div>
    </main>
    <main>
      <div>
        <h1
          className="ml-24 text-3xl font-medium"
        >
          Most Viewed
        </h1>
      </div>
    </main>
    <main>
      <div>
        <h1
          className="ml-24 text-3xl font-medium"
        >
          Recommended movies
        </h1>
      </div>
    </main>
    <main>
      <div>
        <h1
          className="ml-24 text-3xl font-medium"
        >
          Recommended movies
        </h1>
      </div>
    </main>
    <main>
      <div>
        <h1
          className="ml-24 text-3xl font-medium"
        >
          Recommended movies
        </h1>
      </div>
    </main>
  </div>,
]
`;
